synapse/rest/client/v2_alpha/report_event.py

```diff
@@ -31,16 +44,39 @@
 class ReportEventRestServlet(RestServlet):
     PATTERNS = client_patterns("/rooms/(?P<room_id>[^/]*)/report/(?P<event_id>[^/]*)$")
 
+    # As we cannot import HomeServer without creating a circular dependency,
+    # use a forward reference.
+    hs: 'HomeServer'
+    auth: Auth
+    federation_sender: Optional[FederationSender]
+    server_notices: ServerNoticesManager
+    state: StateHandler
+    storage: Storage
+    store: DataStore
```

Comment:
this should be unnecessary - annotate the `hs` param to `__init__` and mypy/pycharm should figure out the rest.

(it's also not valid syntax in python 3.5, though since this is unlikely to land before we drop support for py3.5, that's probably unimportant.